api/jobs/queue.py

```diff
@@ -360,30 +360,53 @@ def search(containers, states=None, tags=None):
         ])
 
     @staticmethod
-    def get_statistics():
+    def get_statistics(tags=None, last=None, unique=False, all_flag=False):
         """
         Return a variety of interesting information about the job queue.
         """
 
-        # Count jobs by state
-        result = config.db.jobs.aggregate([{"$group": {"_id": "$state", "count": {"$sum": 1}}}])
-        # Map mongo result to a useful object
+        if all_flag:
+            unique = True
+            if last is None:
+                last = 3
+
+        results = { }
+        match = { } # match all jobs
+
+        if tags is not None and len(tags) > 0:
+            match = { 'tags': {'$in': tags } } # match only jobs with given tags
+
+        # Count jobs by state, mapping the mongo result to a useful object
+        result = list(config.db.jobs.aggregate([{'$match': match }, {'$group': {'_id': '$state', 'count': {'$sum': 1}}}]))
         by_state = {s: 0 for s in JOB_STATES}
         by_state.update({r['_id']: r['count'] for r in result})
+        results['states'] = by_state
+
+        # List unique tags
+        if unique:
+            results['unique'] = sorted(config.db.jobs.distinct('tags'))
+
+        # List recently modified jobs for each state
+        if last is not None:
+            results['recent'] = {s: config.db.jobs.find({'$and': [match, {'state': s}]}, {'modified':1}).sort([('modified', pymongo.DESCENDING)]).limit(last) for s in JOB_STATES}
```

Comment:
Rather than doing N (job state) `find`s we could use a mongo aggregation pipeline that does a sort, followed by a group (by state), followed by a limit. I'm not convinced that would be faster on larger datasets but it's something to think about if this ends up being a problem.